tasks/request_files/request_files.py

```diff
@@ -138,19 +137,26 @@ def task(
         )
         retry_sleep_secs = DEFAULT_RESTORE_RETRY_SLEEP_SECS
 
-    # Get retrieval type
-    try:
-        retrieval_type = os.environ[OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY]
-        if retrieval_type not in ("Standard", "Bulk", "Expedited"):
-            msg = (
-                f"Invalid RESTORE_RETRIEVAL_TYPE: '{retrieval_type}'"
-                " defaulting to 'Standard'"
-            )
-            LOGGER.info(msg)
-            retrieval_type = DEFAULT_RESTORE_RETRIEVAL_TYPE
-    except KeyError:
-        LOGGER.warn("Invalid RESTORE_RETRIEVAL_TYPE: 'None' defaulting to 'Standard'")
+    VALID_RESTORE_TYPES = ["Bulk", "Expedited", "Standard"] 
+
+    # Get config and env set to None if the key does not exist
+    env_retrieval_type = os.getenv(OS_ENVIRON_RESTORE_RETRIEVAL_TYPE_KEY, None)
+    config_retrieval_type = event["config"].get(CONFIG_RESTORE_DEFAULT_RETRIEVAL_TYPE_OVERRIDE_KEY, None)
+    if env_retrieval_type in VALID_RESTORE_TYPES:
```

Comment:
Why are we checking this here? It may not be needed, and it only logs when there isn't a problem.

Reply:
this is added to check that the value is getting from the env variable and is also need to verify some unit test in case of  valid env variable.

Reply:
By that logic, would you not want to add logging statements verifying that the config and default values are valid as well? And is it helpful to only log when there are no issues?

Unit tests should be checking behavior, such as making sure that when env_retieval_type is used, it rejects bad values and accepts good ones. Checking an `info` statement is not valid, since it does not verify the ultimate behavior, just that the value was theoretically accepted at one point; code after this point could mess with that value or fail to use it when appropriate.

Reply:
added more logging and updated unit tests that check the retrieval_type actual and expected value